src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -182,6 +188,14 @@ private static uint HandleEventShutdownComplete(State state, ref ConnectionEvent
 
             // Stop accepting new streams.
             state.AcceptQueue.Writer.Complete();
+
+            // Stop notifying about available streams.
+            lock (state)
+            {
+                state.NewUnidirectionalStreamsAvailable?.SetException(ExceptionDispatchInfo.SetCurrentStackTrace(new QuicOperationAbortedException()));
+                state.NewBidirectionalStreamsAvailable?.SetException(ExceptionDispatchInfo.SetCurrentStackTrace(new QuicOperationAbortedException()));
+            }
```

Comment:
Probably want to set these to null just in case.

Reply:
My original thinking was that the exception is the final state of the TCS and than is somebody calls `WaitForAvailable*StreamsAsync`, they would get this exception.
However, I already have a check in place for that:
https://github.com/ManickaP/runtime/blob/78b526da7f2460f90b7936e7044f5519ef8312f8/src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs#L406-L409

So I decided to go with the suggestion.